### ISSUE TITLE
Extending the power of R to a wider audience in developing countries

### ISSUE THREAD

**Issue by Emaasit** [ropensci/unconf16#12]
Before I came to the US for graduate school, I had **never** heard of the R programming language. Many statisticians/data analysts in Africa especially use proprietary software like SAS or SPSS. Since learning R, I have tried to extend the same to others in Africa by creating eight R Meetup groups [(datascience-africa.org).](http://datascience-africa.org/) However, there's still low enthusiasm as people prefer the "point-and-click" type software. I would like to find out from participants the following:
1. What are some of the techniques you have used to extend R usage and moral within your company or University?
2. What are some of the effective ways of teaching R to non-programmers?
3. Since internet usage is limited and costly in developing countries, are there packages that let you use R offline, especially for managing packages?

**Comment by maelle** [comment]
I find it interesting too.

In my previous institution there was an international training programm for field epidemiologists, where statistics were taught using Stata.

**Comment by jennybc** [comment]
@Emaasit Do you think there's a big difference in access or inclination to use information on the internet, i.e. R tutorials, twitter, stack overflow, etc.? I realize this doesn't directly address the preference for point-and-click but does address the lonely agony of learning a new thing.

**Comment by Emaasit** [comment]
@jennybc That's right. The issue is two-fold. Firstly, internet usage is limited & quite costly if used for on-line learning. Secondly, since R is not taught in many Universities in Africa, students are less inclined to research about it and download online materials.

**Comment by Emaasit** [comment]
@masalmon Was there a particular reason for the use of Stata in preference to R in your school in Barcelona?

**Comment by maelle** [comment]
It was in Berlin. I honestly have no idea. It seems to be a tradition. In France in my MSc we learnt R but in my MPH we used Stata so maybe that's an epidemiology thing. In my current institutio  both Stata and R are used. I guess Stata is easier at first glance?

Sometimes I wonder if an argument could be = give away less money for softwares and more money for statisticians but I might be biased. ☺

As regards your 1st question, showing users of other softwares all you can do in R? And for 2), promising support to newbies, i.e. saying they can ask anything so that they feel "safe"?

**Comment by jennybc** [comment]
There's also a generational and inertia thing. The people senior enough to be setting curriculum and teaching are often stuck in the late 1990s, when they set up their stat computing toolkit. To be more sympathetic, they probably don't have the time to burn everything down and start over.

**Comment by leeper** [comment]
Stata is extremely popular in Economics and to some extent in Political Science (it is quickly replacing SPSS as the default tool). Aside from point-and-click it also has a very easy command line (for basic stuff) and has some very nice features. I think there is a lot we could emulate from Stata in R in order to make it more attractive to its current user base.

**Comment by maelle** [comment]
Can you give a few examples of these nice features, @leeper?

**Comment by seankross** [comment]
@Emaasit This topic is of particular interest to me. A handful of volunteers have been translating some of the course materials for [swirl](https://github.com/swirldev/swirl) into languages other than English. @wush978 (who is one of swirl's contributors) teaches a great [R course](https://github.com/wush978/DataScienceAndR) at National Taiwan University.

**Comment by maelle** [comment]
Oh that's cool @seankross , how are these volunteers organized? I might want to help for French translations a little!

**Comment by seankross** [comment]
@masalmon it's organized mainly around a [GitHub repo](https://github.com/swirldev/translations) and a [Google group](https://groups.google.com/forum/#!forum/swirl-translators). If you have any questions about getting involved please let me know! We currently have swirl's menus translated into French ([here](https://github.com/swirldev/translations/blob/master/menus/french.yaml)) but no courses yet. The courses reside [here](https://github.com/swirldev/swirl_courses).

**Comment by Emaasit** [comment]
@seankross That's a great initiative. Another popular language is Swahili, spoken in East & Central Africa. Although the language of instruction in schools in that region is English, natives prefer communicating in the local Swahili language. I am actually know a friend who writes R blogs in swahili. I will ask him to volunteer.

**Comment by seankross** [comment]
@Emaasit Sounds good! If he prefers to communicate over email: info@swirlstats.com

**Comment by maelle** [comment]
@seankross right now I don't have much time but I'd definitely be glad (in a few weeks) to help translate the courses into French if you think there would be people using the French versions!

**Comment by benmarwick** [comment]
I've taught R to archaeologists in Thailand, Indonesia and Myanmar (I wrote a note about that here: http://software-carpentry.org/blog/2015/03/teaching-in-yangon.html). To directly answer the questions,  teaching techniques pioneered by Software Carpentry have been highly successful for me. Lots of live coding and sticky notes are the key ingredients. I usually modify the lesson content to make it simpler and more domain specific. The [minicran](https://github.com/RevolutionAnalytics/miniCRAN) package has been a life saver for distributing packages to learners in offline and low bandwidth environments. 

One the challenge of uptake, where academic culture is highly conservative (especially Thailand in my experience) uptake is slow because students are reluctant to do something that they know their professors don't understand. So that's a bigger challenge that requires generational turnover.

**Comment by Emaasit** [comment]
@benmarwick That's a great blog post!!!! Minicran is exactly what I have been looking for. I shall explore how it works and immediately recommend it to my friends back home.

**Comment by lmguzman** [comment]
I taught R to ecologist in Brazil, and helped starting a study group in Portuguese. I find one of the most challenging things is not only the translation but to help them keeping the energy going. They were very keen to translate the study group website, but since they don't have large support groups like we do, I feel they have slowed down lately.

**Comment by jules32** [comment]
Hi there, I'd be interested in this discussion too. We require R/RStudio/Github for our [Ocean Health Index project](HTTP://ohi-science.org), which requires a lot of scientific and policy expertise on top of open science tools. We work with a wide range of countries and governments and getting them comfortable with these open science tools and their power as quickly as possible is important so they can do their work. We are beginning to provide more support for learning these tools, but are excited to connect with others about this!

**Comment by gdickens** [comment]
@benmarwick thanks for the blog post and @Emaasit thanks for starting the topic. 

I'm living in Mynanmar and about to start teaching a group how to use R so all this information is extremely useful. 

The idea of translating the Swirl courses into other languages is also great.
